src/compile/scale/properties.ts

```diff
@@ -418,18 +422,21 @@ export function zero(
     }
   }
 
-  // If there is no custom domain, return true only for the following cases:
+  const defaultZero = scaleConfig?.zero === undefined ? true : scaleConfig?.zero;
```

Comment:
Instead of adding this line, probably cleaner to add `zero: true` to `defaultScaleConfig`.